flang/include/flang/Optimizer/CodeGen/CGOps.td

```diff
@@ -91,7 +91,8 @@ def fircg_XArrayCoorOp : fircg_Op<"ext_array_coor", [AttrSizedOperandSegments]>
     be converted to an extended embox. This op will have the following sets of
     arguments.
 
-       - memref: The memory reference of the array's data.
+       - memref: The memory reference of the array's data. It can be a fir.box if
+         the underlying data is not contiguous.
```

Comment:
I'm still curious if it wouldn't make some sense to have 2 distinct code gen ops. Since the CG ops are of super limited scope and use, having two ops could eliminate all the conditional rewriting logic, it seems.

Reply:
To me, the complexity of ArrayCoorOp lies in the lower-bounds/triplets/subcomponents part, not really in the stride computation. I feel having two ops would just duplicate the complex part, with potential bugs coming from that. So I am more incline to stay small in code footprint for now. But yes, it adds a bit of additional complexity to it. When dust settles, revisiting this decision with maybe some other motivations for splitting the path will make some sense to me.